File: client/java/src/avt/RegisterScr.java
package avt;

import java.util.Vector;
import javax.microedition.lcdui.Graphics;
import main.Canvas;
import main.GameMidlet;

public final class RegisterScr extends MyScreen {
   private static RegisterScr instance;
   private byte male = 1;
   public int index = 0;
   private int selected;
   public int countLeft;
   public int countRight;
   private Vector listHair;
   private Vector listClothing;
   private Vector listQ;
   private int time = 0;

   public static RegisterScr gI() {
      if (instance == null) {
         instance = new RegisterScr();
      }

      return instance;
   }

   public final void commandTab(int var1, int var2) {
      switch (var1) {
         case 0:
            Vector var3;
            (var3 = new Vector()).addElement(new Command(T.yes, 0, this));
            var3.addElement(new Command(T.no, 1, this));
            Canvas.setInfoC(T.milk, var3);
         default:
      }
   }

   public final void commandActionPointer(int var1) {
      switch (var1) {
         case 0:
            doFinish();
            return;
         case 1:
            doFinish();
         default:
      }
   }

   public final void switchToMe() {
      GameMidlet.avatar.direct = 0;
      GameMidlet.avatar.seriPart = new Vector();
      this.getAvatarPart();
      super.center = new Command(T.success, 0);
      SeriPart var1 = new SeriPart();
      int var2 = CRes.random.nextInt(this.listQ.size());
      var1.idPart = ((APartInfo)this.listQ.elementAt(var2)).IDPart;
      GameMidlet.avatar.addSeri(var1);
      var1 = new SeriPart();
      var2 = CRes.random.nextInt(this.listClothing.size());
      var1.idPart = ((APartInfo)this.listClothing.elementAt(var2)).IDPart;
      GameMidlet.avatar.addSeri(var1);
      (var1 = new SeriPart()).idPart = 4;
      GameMidlet.avatar.addSeri(var1);
      var1 = new SeriPart();
      var2 = CRes.random.nextInt(this.listHair.size());
      var1.idPart = ((APartInfo)this.listHair.elementAt(var2)).IDPart;
      GameMidlet.avatar.addSeri(var1);
      GameMidlet.avatar.addSeri(new SeriPart((short)0));
      GameMidlet.avatar.orderSeriesPath();
      PaintPopup.gI().a(T.createChar, 150 * AvMain.hd, 170 + (AvMain.hd == 2 ? 120 : 0), 1);
      super.switchToMe();
   }

   private void getAvatarPart() {
      GameMidlet.avatar.gender = this.male;
      if (this.listHair != null) {
         this.listHair.removeAllElements();
         this.listClothing.removeAllElements();
         this.listQ.removeAllElements();
      }

      this.listHair = new Vector();
      this.listClothing = new Vector();
      this.listQ = new Vector();

      for(int var1 = 0; var1 < AvatarData.listPart.length; ++var1) {
         APartInfo var2;
         if (AvatarData.listPart[var1] instanceof APartInfo && (var2 = (APartInfo)AvatarData.listPart[var1]) != null && (var2.gender == this.male || var2.gender == 0) && var2.level == 0) {
            if (var2.zOrder == 50) {
               this.listHair.addElement(var2);
            } else if (var2.zOrder == 20) {
               this.listClothing.addElement(var2);
            } else if (var2.zOrder == 10) {
               this.listQ.addElement(var2);
            }
         }
      }

      this.selected = 0;
      this.getId();
      if (GameMidlet.avatar.action != 10) {
         GameMidlet.avatar.setAction((byte)1);
      }

      GameMidlet.avatar.orderSeriesPath();
   }

   private static void doFinish() {
      Canvas.isInitChar = true;
      Canvas.startWaitDlg(T.createChar + "...");
      GlobalService.gI().doRequestCreCharacter();
   }

   public final void update() {
      if (this.countLeft > 0) {
         --this.countLeft;
      }

      if (this.countRight > 0) {
         --this.countRight;
      }

      ++this.time;
      if (this.time > 50) {
         this.time = 0;
         int var1 = CRes.random.nextInt(3);
         if (GameMidlet.avatar.action != 10) {
            if (var1 == 0) {
               GameMidlet.avatar.setAction((byte)1);
            } else {
               GameMidlet.avatar.setAction((byte)0);
            }
         }
      }

      GameMidlet.avatar.updateFrame();
   }

   public final void setKeyUpDown(int var1) {
      this.index = var1;
      if (this.index < 0) {
         this.index = 1;
      }

      if (this.index > 1) {
         this.index = 0;
      }

   }

   public final void setKeyLeftRight(int var1) {
      this.selected += var1;
      if (this.selected < 0) {
         this.selected = 1;
      }

      if (this.selected > 1) {
         this.selected = 0;
      }

      if (this.index == 0) {
         if (this.male == 1) {
            this.male = 2;
         } else {
            this.male = 1;
         }

         this.getAvatarPart();
      } else {
         this.getId();
      }

   }

   public final void updateKey() {
      Canvas.paint.initResourceFive();
      super.updateKey();
   }

   private void getId() {
      for(int var1 = 0; var1 < GameMidlet.avatar.seriPart.size(); ++var1) {
         SeriPart var2;
         APartInfo var3;
         if ((var3 = (APartInfo)AvatarData.getPart((var2 = (SeriPart)GameMidlet.avatar.seriPart.elementAt(var1)).idPart)).zOrder == 50 && this.listHair.size() != 0 && this.selected < this.listHair.size()) {
            var2.idPart = ((APartInfo)this.listHair.elementAt(this.selected)).IDPart;
         }

         if (var3.zOrder == 20 && this.listClothing.size() != 0 && this.selected < this.listClothing.size()) {
            var2.idPart = ((APartInfo)this.listClothing.elementAt(this.selected)).IDPart;
         }

         if (var3.zOrder == 10 && this.listQ.size() != 0 && this.selected < this.listQ.size()) {
            var2.idPart = ((APartInfo)this.listQ.elementAt(this.selected)).IDPart;
         }
      }

      GameMidlet.avatar.orderSeriesPath();
   }

   public final void paint(Graphics var1) {
      Canvas.loadMap.paint(var1);
      Canvas.loadMap.paintBackGround(var1);
      Canvas.resetTrans(var1);
      PaintPopup.gI().paint(var1);
      var1.translate(PaintPopup.gI().x, PaintPopup.gI().y);
      Canvas.paint.paintPlayer(var1, this.index, this.male, this.countLeft, this.countRight);
      super.paint(var1);
   }

   public static void onCreaCharacter(boolean var0) {
      Canvas.endDlg();
      if (var0) {
         ClientUtilities.requestChangeZone();
      } else {
         Canvas.startOKDlg(T.createCharFail);
      }

   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/RegisterScr.cs
public class RegisterScr : MyScreen
{
	private class IActionOkUser : IAction
	{
		public void perform()
		{
			instance.doFinish();
		}
	}

	public static RegisterScr instance;

	public static bool isCreateChar;

	public sbyte male = 1;

	public int index;

	public new int selected;

	public int countLeft;

	public int countRight;

	private MyVector listHair;

	private MyVector listClothing;

	private MyVector listQ;

	private int time;

	public static RegisterScr gI()
	{
		if (instance == null)
		{
			instance = new RegisterScr();
		}
		return instance;
	}

	public override void commandTab(int index)
	{
		if (index == 0)
		{
			doFinish();
		}
	}

	public override void switchToMe()
	{
		GameMidlet.avatar.seriPart = new MyVector();
		GameMidlet.avatar.direct = Base.RIGHT;
		getAvatarPart();
		center = new Command(T.success, 0);
		SeriPart seriPart = new SeriPart();
		int i = CRes.r.nextInt(listQ.size());
		seriPart.idPart = ((APartInfo)listQ.elementAt(i)).IDPart;
		GameMidlet.avatar.addSeri(seriPart);
		SeriPart seriPart2 = new SeriPart();
		int i2 = CRes.r.nextInt(listClothing.size());
		seriPart2.idPart = ((APartInfo)listClothing.elementAt(i2)).IDPart;
		GameMidlet.avatar.addSeri(seriPart2);
		SeriPart seriPart3 = new SeriPart();
		seriPart3.idPart = 4;
		GameMidlet.avatar.addSeri(seriPart3);
		SeriPart seriPart4 = new SeriPart();
		int i3 = CRes.r.nextInt(listHair.size());
		seriPart4.idPart = ((APartInfo)listHair.elementAt(i3)).IDPart;
		GameMidlet.avatar.addSeri(seriPart4);
		GameMidlet.avatar.addSeri(new SeriPart(0));
		GameMidlet.avatar.orderSeriesPath();
		init();
		Canvas.paint.initReg();
		base.switchToMe();
	}

	public override void doMenu()
	{
		LoginScr.gI().doMenu();
	}

	public void init()
	{
		PaintPopup.gI().setInfo(T.createChar, 150 * AvMain.hd, 170 + ((AvMain.hd == 2) ? 160 : 0), 1, -1, null, null);
	}

	public void getAvatarPart()
	{
		GameMidlet.avatar.gender = male;
		if (listHair != null)
		{
			listHair.removeAllElements();
			listClothing.removeAllElements();
			listQ.removeAllElements();
		}
		listHair = new MyVector();
		listClothing = new MyVector();
		listQ = new MyVector();
		for (int i = 0; i < AvatarData.listPart.Length; i++)
		{
			if (!(AvatarData.listPart[i] is APartInfo))
			{
				continue;
			}
			APartInfo aPartInfo = (APartInfo)AvatarData.listPart[i];
			if (aPartInfo != null && (aPartInfo.gender == male || aPartInfo.gender == 0) && aPartInfo.level == 0)
			{
				if (aPartInfo.zOrder == 50)
				{
					listHair.addElement(aPartInfo);
				}
				else if (aPartInfo.zOrder == 20)
				{
					listClothing.addElement(aPartInfo);
				}
				else if (aPartInfo.zOrder == 10)
				{
					listQ.addElement(aPartInfo);
				}
			}
		}
		selected = 0;
		getId();
		GameMidlet.avatar.orderSeriesPath();
		if (GameMidlet.avatar.action != 10)
		{
			GameMidlet.avatar.setAction(1);
		}
	}

	protected void doFinish()
	{
		Canvas.isInitChar = true;
		Canvas.startWaitDlg(T.createChar + "...");
		GlobalService.gI().doRequestCreCharacter();
	}

	public override void keyPress(int keyCode)
	{
		base.keyPress(keyCode);
	}

	public override void update()
	{
		if (countLeft > 0)
		{
			countLeft--;
		}
		if (countRight > 0)
		{
			countRight--;
		}
		time++;
		if (time > 50)
		{
			time = 0;
			int num = CRes.r.nextInt(3);
			if (GameMidlet.avatar.action != 10)
			{
				if (num == 0)
				{
					GameMidlet.avatar.setAction(1);
				}
				else
				{
					GameMidlet.avatar.setAction(0);
				}
			}
		}
		GameMidlet.avatar.updateAvatar();
	}

	public void setKeyUpDown(int ind)
	{
		index = ind;
		if (index < 0)
		{
			index = 1;
		}
		if (index > 1)
		{
			index = 0;
		}
	}

	public void setKeyLeftRight(int ind)
	{
		selected += ind;
		if (selected < 0)
		{
			selected = 1;
		}
		if (selected > 1)
		{
			selected = 0;
		}
		if (index == 0)
		{
			if (male == 1)
			{
				male = 2;
			}
			else
			{
				male = 1;
			}
			getAvatarPart();
		}
		else
		{
			getId();
		}
	}

	public override void updateKey()
	{
		Canvas.paint.updateKeyRegister();
		base.updateKey();
	}

	private void getId()
	{
		for (int i = 0; i < GameMidlet.avatar.seriPart.size(); i++)
		{
			SeriPart seriPart = (SeriPart)GameMidlet.avatar.seriPart.elementAt(i);
			APartInfo aPartInfo = (APartInfo)AvatarData.getPart(seriPart.idPart);
			if (aPartInfo.zOrder == 50 && listHair.size() != 0 && selected < listHair.size())
			{
				seriPart.idPart = ((APartInfo)listHair.elementAt(selected)).IDPart;
			}
			if (aPartInfo.zOrder == 20 && listClothing.size() != 0 && selected < listClothing.size())
			{
				seriPart.idPart = ((APartInfo)listClothing.elementAt(selected)).IDPart;
			}
			if (aPartInfo.zOrder == 10 && listQ.size() != 0 && selected < listQ.size())
			{
				seriPart.idPart = ((APartInfo)listQ.elementAt(selected)).IDPart;
			}
		}
		GameMidlet.avatar.orderSeriesPath();
	}

	public override void paint(MyGraphics g)
	{
		Canvas.loadMap.paint(g);
		Canvas.loadMap.paintObject(g);
		Canvas.resetTrans(g);
		PaintPopup.gI().paint(g);
		Canvas.resetTrans(g);
		g.translate(PaintPopup.gI().x, PaintPopup.gI().y);
		Canvas.paint.paintPlayer(g, index, male, countLeft, countRight);
		base.paint(g);
	}

	public void onCreaCharacter(bool isCreaCha)
	{
		Canvas.endDlg();
		if (isCreaCha)
		{
			MapScr.gI().joinCitymap();
		}
		else
		{
			Canvas.startOKDlg(T.createCharFail);
		}
	}

	public void onRegister(string userName, string pass)
	{
		Canvas.user = userName;
		Canvas.pass = pass;
		AvatarData.saveMyAccount();
		GlobalMessageHandler.gI().miniGameMessageHandler = null;
		ServerListScr.gI().login();
	}
}
